tests: subsys/debug/coredump: update for CM33

hakehuang opened the pull request:
update test address for CM33 platform

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

nashif commented on tests/subsys/debug/coredump/src/main.c:
```diff
@@ -26,6 +26,10 @@ void func_1(uint32_t *addr)
 	printk("Coredump: %s\n", CONFIG_BOARD);
-
+#ifdef CONFIG_BOARD_MIMXRT685_EVK
+	/* for CM33 0x0 is sharememory */
```
shared memory?

nashif replied:
btw, this would probably not work for many other boards, do not want to see things being ifdefed here, we either exclude such platforms or implement this is more scalable way.

hakehuang replied:
@nashif , from the CM33 [Technical document](https://developer.arm.com/documentation/100230/0003/functional-description/programmers-model/memory-model?lang=en), 0x0 to 0x1FFFFFFF is code area, but for NXP platform, we separated this ware as
![image](https://user-images.githubusercontent.com/800198/93417316-5f151e80-f8da-11ea-846c-e95f049b7de3.png)

so this is really a platoform dependant, I update the test address to modify the main directly,please check.

dcpleung reviewed (changes requested):
Now you are changing behavior on other platforms where it is no longer causing null pointer reference. Trying to set the address of `main()` to zero is not going to cause fatal error if text resides in RAM and that memory is marked RW (read and write).

hakehuang committed:
test: subsys/debug/coredump exclude mimxrt685_evk_cm33

this case use a NULL pointer to test, however, mimxrt685 0 address
is overlapped with 0x30000000 which is used as SRAM. write to this
address will cause unpredicted issue, in this case, no console output.
The printk data is corrupted. so exclude this platform.

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
> Now you are changing behavior on other platforms where it is no longer causing null pointer reference. Trying to set the address of `main()` to zero is not going to cause fatal error if text resides in RAM and that memory is marked RW (read and write).

@dcpleung thanks I revert this change and just exclude mimxrt685_evk_cm33 platform, the issue in this platform is that the 0 address is overlapped with SRAM address 0x30000000, which unfortunately keeps some key data for printk.

hakehuang commented:
this issue can be fixed by #28543 , so I close this PR. Thanks all for review.
